src/OpenSpaceToolkit/Astrodynamics/NumericalSolver.cpp

```diff
@@ -518,11 +518,23 @@ NumericalSolver::ConditionSolution NumericalSolver::integrateDuration(
         return isSatisfied ? 1.0 : -1.0;
     };
 
+    // If the condition is satisfied at both previousTime and currentTime, then the condition
+    // was satisfied at the start of the integration
```

Comment:
Huuum, no okay I think I see:
- We go through the `while` loop and integrate. In there, if at some point, the condition is satisfied we would get out of the loop and then have: condition not satisfied at previous time and satisfied at current time.
- So if we end up with previous time and current time satisfying the condition it is because we did not go through the while loop and condition was satisfied at the beginning

Reply:
Is there any value in adding that check before the while loop code? 🤔 We'd have to take a first integration step I guess?